### PR TITLE
check_alsabat.sh: add format support

### PR DESCRIPTION
**Pull request by keqiaozhang**
add a -f option to support audio format.

### CODE REVIEW

**Review by dbaluta — DISMISSED** [review]

File: test-case/check-alsabat.sh [hunk 1]
```diff
@@ -25,22 +25,25 @@ OPT_NAME['p']='pcm_p'     	OPT_DESC['p']='pcm for playback. Example: hw:0,0'
 OPT_HAS_ARG['p']=1          	OPT_VAL['p']=''
 
 OPT_NAME['C']='channel_c'       OPT_DESC['C']='channel number for capture.'
-OPT_HAS_ARG['C']=1             OPT_VAL['C']='1'
+OPT_HAS_ARG['C']=1              OPT_VAL['C']='1'
 
 OPT_NAME['r']='rate'            OPT_DESC['r']='sample rate'
-OPT_HAS_ARG['r']=1             OPT_VAL['r']=48000
+OPT_HAS_ARG['r']=1              OPT_VAL['r']=48000
```
Comment (marc-hb) [inline test-case/check-alsabat.sh:31]: This is pure git noise, don't.
https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes

**Review by marc-hb — CHANGES_REQUESTED** [review]
This PR is hiding a backwards-incompatible change. I understand you did that to align with `alsabat` which is good but it will break existing users. This should at the very least be explained in a separate commit message, clearly visible in the git log.

- commit 1: rename -f[requency] option to -F for alsabat consistency
- commit 2: add new -f[format] option

With the NEW code, what happens when someone does `check-alsabat.sh -f 821`? Do they get a meaningful error message from alsabat? Add that error message in the first commit message.

File: test-case/check-alsabat.sh (hunk 1 above)
Comment (keqiaozhang) [reply test-case/check-alsabat.sh]: Updated.

**Comment by keqiaozhang** [comment]
Please help to review. If no objections, I will merge this PR myself, because I also need to update the parameters(`-f => -F`) in web side for alsabat related cases.

image

File: test-case/check-alsabat.sh [hunk 2]
```diff
@@ -33,8 +33,11 @@ OPT_HAS_ARG['r']=1             OPT_VAL['r']=48000
 OPT_NAME['c']='pcm_c'      	OPT_DESC['c']='pcm for capture. Example: hw:1,0'
 OPT_HAS_ARG['c']=1             OPT_VAL['c']=''
 
-OPT_NAME['f']='frequency'       OPT_DESC['f']='target frequency'
-OPT_HAS_ARG['f']=1             OPT_VAL['f']=821
+OPT_NAME['f']='format'       OPT_DESC['f']='target format'
+OPT_HAS_ARG['f']=1             OPT_VAL['f']="S16_LE"
+
+OPT_NAME['F']='frequency'       OPT_DESC['F']='target frequency'
+OPT_HAS_ARG['F']=1             OPT_VAL['F']=821
```
Comment (miRoox) [inline test-case/check-alsabat.sh:40]: Will the change of `-f` break the backward compatibility? For example, daily MTL test runs for different frequencies with `-f`.

Comment (keqiaozhang) [reply test-case/check-alsabat.sh:40]: Yes, so we need to update the alsabat test config in web accordingly when merging this PR.

**Review by marc-hb — APPROVED** [review]

**Comment by marc-hb** [comment]
Very small conflict with #991 that I just merged, need to rebase sorry.

**Comment by keqiaozhang** [comment]
@marc-hb , rebased. Please give a check again. if no problems, I will merge it today.

**Review by marc-hb — APPROVED** [review]
Please make sure the test results are OK-ish.

**Pull request merged**